core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -128,23 +133,31 @@ public String toString() {
     private final Path sharedDataPath;
     private final Lock[] locks;
 
-    private final boolean addNodeId;
+    private final boolean addLockIdToCustomPath;
 
-    private final int localNodeId;
+    private final int nodeLockId;
     private final AtomicBoolean closed = new AtomicBoolean(false);
     private final Map<ShardId, InternalShardLock> shardLocks = new HashMap<>();
 
+    private final NodeMetaData nodeMetaData;
+
     /**
      * Maximum number of data nodes that should run in an environment.
      */
     public static final Setting<Integer> MAX_LOCAL_STORAGE_NODES_SETTING = Setting.intSetting("node.max_local_storage_nodes", 50, 1,
             Property.NodeScope);
 
     /**
-     * If true automatically append node id to custom data paths.
+     * If true automatically append node lock id to custom data paths.
      */
-    public static final Setting<Boolean> ADD_NODE_ID_TO_CUSTOM_PATH =
-        Setting.boolSetting("node.add_id_to_custom_path", true, Property.NodeScope);
+    public static final Setting<Boolean> ADD_NODE_LOCK_ID_TO_CUSTOM_PATH =
+        Setting.boolSetting("node.add_lock_id_to_custom_path", true, Property.NodeScope);
+
+
+    public static final Setting<Long> NODE_ID_SEED_SETTING =
+        // don't use node.id.seed so it won't be seen as an attribute
```

Comment:
I vaguely remember that attributes have been moved to attrs. namespace? Can this finally become `node.id.seed`?